src/main/java/com/example/demo/dao/GradeLocalRepository.java

```diff
@@ -0,0 +1,73 @@
+package com.example.demo.dao;
+
+import com.example.demo.dao.entity.Grade;
+import org.springframework.stereotype.Repository;
+
+import java.util.*;
+
+@Repository
+public class GradeLocalRepository implements GradeRepository {
+    private Map<Long, List<Grade>> gradeRegister;
```

Comment:
ta nazwa nic nie mówi. 

`studentIdToGrade` jak troche poszukasz jak nazywać mapy w javie, to taka konwencja mysle że będize na prowadzeniu. 

czyli wiadomo od razu, że kluczem jest studentId a wartością Grade